lib/rbUtil.js

```diff
@@ -239,15 +239,19 @@ function HTTPError(response) {
     if (response.body && response.body.type) {
         this.message += ': ' + response.body.type;
     }
-
-    for (var key in response) {
-        this[key] = response[key];
-    }
+    Object.assign(this, response);
 }
 util.inherits(HTTPError, Error);
 
 rbUtil.HTTPError = HTTPError;
 
+function HTTPRedirect(response) {
```

Comment:
> but also assumes that code several layers deep knows about how a response is going to be used, which it probably shouldn't need to know & care about.

I'm not sure what do you mean. The only change needed to support error-like redirects is adding 1 line in server to treat the redirect same way as HTTPError.

If we make a redirect normal response, we'll need to add A LOT of checks, as throwing a redirect instead of returning it, we can shortcut really deep promise chains. I've tried to add all those checks today,and the code doesn't seem to become more cleaner.

If we make a decision to switch to `Response`-like style in RB, we'd need to change the behaviour of `HTTPError` too, so `HTTPRedirect` would be a minor issue.